### PR TITLE
CA1860: Avoid using 'Enumerable.Any()' extension method

### PR DESCRIPTION
**Pull request by Evangelink**
Fixes newly enabled .NET analyzers CA1860 rule. More information about the rule at https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1860

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by Evangelink** [comment]
Sorry not sure what happened, I had no error locally but now I do reproduce the same errors. Will investigate and mark as ready when build is passing locally.

File: Src/FluentAssertions/Execution/AssertionScope.cs [hunk 1]
```diff
@@ -169,7 +169,7 @@ public AssertionScope BecauseOf(string because, params object[] becauseArgs)
             {
                 string becauseOrEmpty = because ?? string.Empty;
 
-                return becauseArgs?.Any() == true
+                return becauseArgs?.Length != 0
```
Comment (jnyrup) [inline Src/FluentAssertions/Execution/AssertionScope.cs:172]: This is not an equivalent rewrite when `becauseArgs` is null

Comment (jnyrup) [reply Src/FluentAssertions/Execution/AssertionScope.cs:172]: but `becauseArgs?.Length > 0` is

**Comment by jnyrup** [comment]
> Fixes newly enabled .NET analyzers CA1860 rule.

Since we don't have very performance sensitive code, I don't have a strong opinion on whether or not we should limit the usage of `Any`. 
But if we do, we should stricten the analyzer to enforce it.

**Comment by dennisdoomen** [comment]
> Since we don't have very performance sensitive code, I don't have a strong opinion on whether or not we should limit the usage of `Any`. But if we do, we should stricten the analyzer to enforce it.

It depends on the run-time version. As of .NET Core 2.0, `Any` will be check whether you're calling it on an array or collection and directly use the `Count` property. The .NET 4.x version isn't that smart.  

But I agree. Totally fine to enable this rule.

**Comment by jnyrup** [comment]
Thinking more about it, I'm pro enforcing this. 
Limiting the usage of `Any` should help spotting places where we do multiple enumerations on lazy non-materialized enumerables.

Related analyzer and issue:
https://github.com/fluentassertions/fluentassertions/blob/22a603980d977ee37ef0f3c5a9bcdffec1f2e1d5/.editorconfig#L153-L154

**Comment by Evangelink** [comment]
Will fix this PR and enable the rule today. Sorry for the delay, I was held back by some other tasks.

**Comment by Evangelink** [comment]
Sorry for the delay here and multiple commits.... Hope this PR will help!

Thanks for the awesome tool :)

**Review by jnyrup — APPROVED** [review]

**Review by dennisdoomen — APPROVED** [review]

**Pull request merged**